src/agents/der_aggregator.jl

```diff
@@ -0,0 +1,636 @@
+            @variable(DERA_WM, dera_discharge[model_data.index_d, model_data.index_t] >= 0)
+            @variable(DERA_WM, dera_energy[model_data.index_d, model_data.index_t] >= 0)
+            @variable(DERA_WM, dera_pv_gen[model_data.index_d, model_data.index_t] >= 0)
+
+            dera_stor_capacity_h =
+                h -> begin
+
+                (incentive_function_intercept[i] + incentive_function_slope[i] * incentive) * total_der_stor_capacity(z, h)
+
+            end
+
+            dera_stor_capacity =
+                begin
+
+                sum(dera_stor_capacity_h(h) for h in model_data.index_h)
+
+            end
+
+            # when it comes to the aggregated pv capacity, use aggregated storage capacity, divide by optimal storage size to get the number of households that's participating the dera aggregation,
+            # then multiply by the optimal pv size,
+            # TODO: I'm using optimal storage and pv capacities for existing resources, need to be more flexible to account for changing optimal tech sizes overtime.
+            # right now, assume the optimal sizes are the same for both existing and new techs.
+            dera_pv_capacity_h = 
+                h -> begin
+
+                dera_stor_capacity_h(h) / customers.Opti_DG_E(z, h, :BTMStorage) * customers.Opti_DG_E(z, h, :BTMPV)
+
+            end
+
+            dera_pv_capacity =
+                begin
+
+                sum(dera_pv_capacity_h(h) for h in model_data.index_h)
+
+            end
+
+            objective_revenue = begin
+                sum(
+                    model_data.omega(d) * delta_t *
+                    ipp.LMP_my(reg_year_index, z, d, t) *
+                    (dera_discharge[d, t] - dera_charge[d, t] + dera_pv_gen[d, t]) 
+                    for d in model_data.index_d, t in model_data.index_t
+                ) + ipp.capacity_price(reg_year_index) * (dera_stor_capacity .* ipp.capacity_credit_stor_E_my(reg_year_index, z, Symbol("der_aggregator")) + 
+                dera_pv_capacity .* ipp.capacity_credit_E_my(reg_year_index, z, Symbol("dera_pv")))
+            end
+
+            objective_cost = begin
+                incentive * dera_stor_capacity
+            end
+
+            @objective(DERA_WM, Max, objective_revenue - objective_cost)
+
+            # DERA (storage) constraints -- We assume all distributed storage resources have the same initial energy, duration and round-trip-efficiency
+
+            @constraint(
+                DERA_WM,
+                Eq_stor_energy_balance[
+                    d in model_data.index_d,
+                    t in model_data.index_t.elements[2:end],
+                ],
+                dera_energy[d, t] == dera_energy[d, model_data.index_t.elements[findall(x -> x == (model_data.time(t)-delta_t), model_data.time.values)][1]] - dera_discharge[d, t] / customers.rte_dist_stor(z, :Commercial) * delta_t +
```

Comment:
Why is this calling out `rte_dist_stor` for `:Commercial`?

Reply:
I think I just assumed a random round-trip-efficiency number here. What we really need here is the round-trip-efficiency of DER aggregator.

Reply:
Issue

Reply:
#10